File: Python/0048_rotate_image/solution.py
def rotate_image(matrix):
    copy = []
    length = 0
    while len(copy) != len(matrix):
        curr = []
        for index in range(len(matrix)-1, -1, -1):
            curr.append(matrix[index][length])
        copy.append(curr)
        length += 1
    for i in range(len(copy)):
        for j in range(len(copy[i])):
            matrix[i][j] = copy[i][j]
    return matrix

matrix = [[5,1,9,11],[2,4,8,10],[13,3,6,7],[15,14,12,16]]
print(rotate_image(matrix))

File: JavaScript/0048_rotate_image/solution.js
function rotate_image(matrix) {
    let copy = [];
    let _length = 0;
    while (copy.length != matrix.length) {
        let curr = [];
        for (let index = matrix.length -1; index >= 0; index --) {
            curr.push(matrix[index][_length]);
        }
        copy.push(curr);
        _length ++;
    }
    for (let i = 0; i < copy.length; i ++) {
        for (let j = 0; j < copy[i].length; j ++) {
            matrix[i][j] = copy[i][j];
        }
    }
    return matrix;
}

let matrix = [[5,1,9,11],[2,4,8,10],[13,3,6,7],[15,14,12,16]];
console.log(rotate_image(matrix));``